### PR TITLE
planner: support wildcards for auto-capture black-list

### LINKED ISSUE #32714: planner: support wildcards for auto-capture black-list
## Enhancement
support wildcards for auto-capture black-list

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #32714

Problem Summary: planner: support wildcards for auto-capture black-list

### What is changed and how it works?

planner: support wildcards for auto-capture black-list

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
planner: support wildcards for auto-capture black-list
```

### CODE REVIEW

File: bindinfo/handle.go [hunk 1]
```diff
@@ -709,19 +707,13 @@ func (h *BindHandle) extractCaptureFilterFromStorage() (filter *captureFilter) {
 		filterTp := strings.ToLower(row.GetString(0))
 		valStr := strings.ToLower(row.GetString(1))
 		switch filterTp {
-		case "db":
-			filter.dbs[valStr] = struct{}{}
```
Comment (qw4990) [inline bindinfo/handle.go:713]: The type `db` is merged into `table`: `db` --> `db.*`.

File: bindinfo/handle.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ import (
 	"sync/atomic"
 	"time"
 
+	tablefilter "github.com/pingcap/tidb-tools/pkg/table-filter"
```
Comment (time-and-fate) [inline bindinfo/handle.go:27]: I take a look at the README of this package and find it supports many features, such as allow/block list and file import. Is that expected?

Comment (qw4990) [reply bindinfo/handle.go:27]: You are right. I've refined related logic and added more tests, PTAL @time-and-fate

File: bindinfo/handle.go [hunk 3]
```diff
@@ -636,6 +636,21 @@ func (cf *captureFilter) isEmpty() bool {
 	return len(cf.tables) == 0 && len(cf.users) == 0
 }
 
+// ParseCaptureTableFilter checks whether this filter is valid and parses it.
+func ParseCaptureTableFilter(tableFilter string) (f tablefilter.Filter, valid bool) {
+	// forbidden wildcards '!' and '@' for safety,
+	// please see https://github.com/pingcap/tidb-tools/tree/master/pkg/table-filter for more details.
+	if strings.ContainsAny(tableFilter, "!@") {
```
Comment (time-and-fate) [inline bindinfo/handle.go:643]: It's possible that a table name contains `!` or `@`.
According to the implementation of `(*tableRulesParser).parse()`, I think we can first `strings.TrimLeft(tableFilter, " \t")`, then check if the first character is `!` or `@`.

File: bindinfo/handle.go [hunk 4]
```diff
@@ -633,14 +633,27 @@ func (cf *captureFilter) Leave(in ast.Node) (out ast.Node, ok bool) {
 }
 
 func (cf *captureFilter) isEmpty() bool {
-	return len(cf.dbs) == 0 && len(cf.tables) == 0 && len(cf.users) == 0
+	return len(cf.tables) == 0 && len(cf.users) == 0
+}
+
+// ParseCaptureTableFilter checks whether this filter is valid and parses it.
+func ParseCaptureTableFilter(tableFilter string) (f tablefilter.Filter, valid bool) {
+	// forbidden wildcards '!' and '@' for safety,
```
Comment (time-and-fate) [inline bindinfo/handle.go:641]:
```suggestion
	// forbid wildcards '!' and '@' for safety,
```

File: bindinfo/handle.go (hunk 3 above)
Comment (qw4990) [reply bindinfo/handle.go]: it has been fixed, PTAL

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**